### PR TITLE
duckdb fixes

### PR DESCRIPTION
**Pull request by ginter**
- duckdb [doesn't support column comments](https://github.com/duckdb/duckdb/discussions/6290)
- ts_or_ds_add wasn't recursively generating the this arg before interpolating

### CODE REVIEW

File: tests/dialects/test_duckdb.py [hunk 1]
```diff
@@ -339,6 +345,12 @@ def test_duckdb(self):
                 "bigquery": "ARRAY_CONCAT([1, 2], [3, 4])",
             },
         )
+        self.validate_all(
+            "SELECT CAST(CAST(x AS DATE) AS DATE) + INTERVAL 1 DAY",
```
Comment (ginter) [inline tests/dialects/test_duckdb.py:349]: the double cast is awkward but there's another test that expects the explicit cast. seems more hacky to try to inspect the rendered sql for 'this' to see if we need to cast again. open to other suggestions...

Comment (barakalon) [reply tests/dialects/test_duckdb.py:349]: checking if `isinstance(this, exp.Cast) and this.is_type(exp.DataType.Type.DATE)` doesn't seem _too_ bad to me.. but I can go either way. 

This is easier with type annotations, but alas, that only happens during optimization.

**Review by barakalon — APPROVED** [review]

File: tests/dialects/test_duckdb.py (hunk 1 above)
Comment (ginter) [reply tests/dialects/test_duckdb.py:349]: it's not that straightforward tho bc we don't know if it's a Cast until we render it at which point we'd have to reparse

Comment (barakalon) [reply tests/dialects/test_duckdb.py:349]: (but that might be overkill)

Comment (tobymao) [reply tests/dialects/test_duckdb.py:349]: this is ok, this optimaziton should go in simplify

**Pull request merged**